### PR TITLE
pkg/system: deprecate SetNamedSecurityInfo, GetSecurityDescriptorDacl

### PR DESCRIPTION
**Pull request by thaJeztah**
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: builder/dockerfile/copy_windows.go [hunk 1]
```diff
@@ -67,21 +67,17 @@ func fixPermissionsWindows(source, destination, SID string) error {
...
-	var dacl *byte
-
-	err = system.GetSecurityDescriptorDacl(&securityDescriptor[0], &daclPresent, &dacl, &daclDefaulted)
+	dacl, _, err := securityDescriptor.DACL()
 	if err != nil {
 		return err
 	}
 
-	return system.SetNamedSecurityInfo(windows.StringToUTF16Ptr(destination), windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, nil, dacl, nil)
+	return windows.SetNamedSecurityInfo(destination, windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, nil, dacl, nil)
```
Comment (thaJeztah) [inline builder/dockerfile/copy_windows.go]: Wondering if `group` and `sacl` should also be set here (previously, they weren't;

```patch
diff --git a/builder/dockerfile/copy_windows.go b/builder/dockerfile/copy_windows.go
index 3f0ea32426..bff7b3aa87 100644
--- a/builder/dockerfile/copy_windows.go
+++ b/builder/dockerfile/copy_windows.go
@@ -76,8 +76,16 @@ func fixPermissionsWindows(source, destination, SID string) error {
        if err != nil {
                return err
        }
+       sacl, _, err := securityDescriptor.SACL()
+       if err != nil {
+               return err
+       }
+       group, _, err := securityDescriptor.Group()
+       if err != nil {
+               return err
+       }

-       return windows.SetNamedSecurityInfo(destination, windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, nil, dacl, nil)
+       return windows.SetNamedSecurityInfo(destination, windows.SE_FILE_OBJECT, windows.OWNER_SECURITY_INFORMATION|windows.DACL_SECURITY_INFORMATION, sid, group, dacl, sacl)
 }

 func validateCopySourcePath(imageSource *imageMount, origPath, platform string) error {
```

**Comment by thaJeztah** [comment]
Failure is a flaky test. Saw we weren't tracking it yet, so opened an issue: https://github.com/moby/moby/issues/40651

```
=== RUN   TestInspect
--- FAIL: TestInspect (12.10s)
    inspect_test.go:39: timeout hit after 10s: waiting for tasks to enter run state. task failed with error: task: non-zero exit (1)
```

**Comment by thaJeztah** [comment]
@lowenna @cpuguy83 @tonistiigi PTAL

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@tiborvass @simonferquel ptal 🤗

File: builder/dockerfile/copy_windows.go (hunk 1 above)
Comment (simonferquel) [reply builder/dockerfile/copy_windows.go:80]: Should not be needed (it will inherit parent objects values by default)

**Review by simonferquel — APPROVED** [review]
lgtm

**Review by tiborvass — APPROVED** [review]

**Pull request merged**